codegen/gateway.go

```diff
@@ -484,6 +489,31 @@ func NewEndpointSpec(
 		}
 	}
 
+	reqHeaderMap := make(map[string]string)
+	m, ok := endpointConfigObj["reqHeaderMap"]
+	if ok {
```

Comment:
Added, but this means these fields are required in the endpoint config. Not sure if that will complicate tchannel endpoints when Matt implements them.